recipes/full_finetune_single_device.py

```diff
@@ -415,9 +416,9 @@ def train(self) -> None:
 
                 # Log peak memory for iteration
                 if self.total_training_steps % self._log_peak_memory_every_n_steps == 0:
-                    log.info(
-                        utils.memory_stats_log("Memory Stats:", device=self._device)
-                    )
+                    memory_stats = utils.memory_stats_log(device=self._device)
+                    self._metric_logger.log_dict(memory_stats, step=self.total_training_steps)
+                    log.info(f"Memory Stats:\n{memory_stats}")
```

Comment:
So right now we log to stdout only when WandB is not enabled, but log to both WandB and stdout when it is enabled? Don't all our metric loggers support log_dict? If so, can we just call `log_dict` only here?